ui/src/papers/index.tsx

```diff
@@ -0,0 +1,210 @@
+    useEffect(() => {
+        (async () => {
+            const allPapers = await getAllPapers();
+            if (allPapers === null) {
+                setViewState({ papers: [], state: ViewState.ERROR });
+                return;
+            }
+            const allPapersWithTitle =
+                await Promise.all(allPapers.map(getPaperInfoFromS2));
+            setViewState({
+                papers: allPapersWithTitle,
+                state: ViewState.READY
+            });
+        })();
+        return () => setViewState({ papers: [], state: ViewState.LOADING });
+    }, []); // The empty array makes the inner callback only execute once
+    return (
+        <>
+            <CssBaseline />
+            <AppBar position="sticky">
+                <Toolbar>
+                    <h1>Supported Papers</h1>
+                </Toolbar>
+            </AppBar>
+            <div>
+                {state === ViewState.LOADING && (
+                    <Box p={4} textAlign="center">
+                        <CircularProgress />
+                    </Box>
+                )}
+                {state === ViewState.READY && (
+                    <Table>
+                        <TableHead>
+                            <TableRow>
+                                <TableCell>Paper</TableCell>
+                                <TableCell>ArXiv ID</TableCell>
+                                <TableCell>Symbols</TableCell>
+                                <TableCell>Citations</TableCell>
+                                <TableCell>Reader Link</TableCell>
+                            </TableRow>
+                        </TableHead>
+                        <TableBody>
+                            {papers.map(paper => (
+                                <TableRow key={paper.s2Id}>
+                                    <TableCell>
+                                        <Typography variant="subtitle1">
+                                            <strong>
+                                                <a href={`https://semanticscholar.org/paper/${paper.s2Id}`}>
+                                                    {paper.title || paper.s2Id}
+                                                </a>
+                                            </strong>
+                                        </Typography>
+                                        {paper.authors.map((author, idx) => (
+                                            <>
+                                                <a key={author.url} href={author.url}>
+                                                    {author.name}
+                                                </a>
+                                                {idx !== paper.authors.length - 1 ? ', ' : null}
+                                            </>
+                                        ))}
+                                        {" "}&bull; {paper.venue}
+                                        {" "}&bull; {paper.year}
```

Comment:
In some cases, I think venue and year might be null. Can we add a check for those here?

Reply:
Good catch, will do.

Reply:
Fixed!